Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditorSettings.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "DocumentPropertyEditor.h"
+
+namespace AzToolsFramework
+{
+    DocumentPropertyEditorSettings::DocumentPropertyEditorSettings(
+        const AZStd::string& settingsRegistryKey,
+        const AZStd::string& propertyEditorName)
+    {
+        m_settingsRegistryBasePath = AZStd::string::format("%s/%s", RootSettingsRegistryPath, propertyEditorName.c_str());
+        m_settingsRegistryDocumentKey = settingsRegistryKey;
+        m_fullSettingsRegistryPath = AZStd::string::format("%s/%s",
+            m_settingsRegistryBasePath.c_str(), m_settingsRegistryDocumentKey.c_str());
+
+        m_fullSettingsFilepath = AZStd::string::format("%s/%s_settings.%s",
+            RootSettingsFilepath, propertyEditorName.c_str(), SettingsRegistrar::SettingsRegistryFileExt);
+
+        m_wereSettingsLoaded = LoadExpanderStates();
+    }
+
+    DocumentPropertyEditorSettings::~DocumentPropertyEditorSettings()
+    {
+        SaveAndCleanExpanderStates();
+    }
+
+    void DocumentPropertyEditorSettings::SaveExpanderStates()
+    {
+        m_settingsRegistrar.StoreObjectSettings(m_fullSettingsRegistryPath, this);
+
+        // Configuration that the dumper util will use when collecting our data from the SettingsRegistry
+        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+        dumperSettings.m_prettifyOutput = true;
+        dumperSettings.m_includeFilter = [pathFilter = m_settingsRegistryBasePath](AZStd::string_view path)
+        {
+            return pathFilter.starts_with(path.substr(0, pathFilter.size()));
```

Comment:
This can use the [SettingsRegistryMergeUtils::IsPathAncestorDescendantOrEqual](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.h#L331-L347) function to determine if the input path is either 1. a prefix of the pathFilter, 2. equal to the pathFilter or 3. the pathFilter is a prefix of the input path.

Reply:
Done.